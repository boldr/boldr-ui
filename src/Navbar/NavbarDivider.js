/* @flow */
import React from 'react';
import classNames from 'classnames';
import { StyleClasses } from '../theme/styleClasses';
import { createWrappedComponent } from '../util/boldrui';

export type Props = {
  tag?: string,
  isBoxed?: boolean,
  className?: string,
};

const BASE_ELEMENT = StyleClasses.NAVBAR_DIVIDER;

export function NavbarDivider({ tag = 'hr', isBoxed, ...props }: Props) {
  const className = classNames(
    BASE_ELEMENT,
    {
      'is-boxed': isBoxed,
    },
    props.className,
  );

  return React.createElement(tag, { ...props, className });
}

export default createWrappedComponent(NavbarDivider);
