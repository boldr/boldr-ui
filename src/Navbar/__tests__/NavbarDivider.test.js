import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { NavbarDivider } from '../NavbarDivider';

describe('NavbarDivider', () => {
  it('should render a hr with .boldr-navbar__divider', () => {
    const wrapper = shallow(<NavbarDivider />);
    expect(wrapper.contains(<hr className="boldr-navbar__divider" />)).toBe(true);
  });

  it('should render a p with .boldr-navbar__divider', () => {
    const wrapper = shallow(<NavbarDivider tag="p" />);
    expect(wrapper.contains(<p className="boldr-navbar__divider" />)).toBe(true);
  });

  it('should render a hr with .boldr-navbar__divider and custom classNames', () => {
    const wrapper = shallow(<NavbarDivider className="custom" />);
    expect(wrapper.hasClass('boldr-navbar__divider')).toBe(true);
    expect(wrapper.hasClass('custom')).toBe(true);
  });
  it('should match the snapshot', () => {
    const wrapper = shallow(<NavbarDivider />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
