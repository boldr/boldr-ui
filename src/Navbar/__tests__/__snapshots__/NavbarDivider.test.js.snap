// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarDivider should match the snapshot 1`] = `
<hr
  className="boldrui-navbar__divider"
/>
`;
